dist/performance.js

```diff
@@ -324,17 +330,53 @@ function getLcpResponseObject(lcpUrl) {
     return responseObject;
 }
 
-function getSpeculationRules() {
-    return Array.from(document.querySelectorAll('script[type=speculationrules]')).map(script => {
-        try {
-            return JSON.parse(script.innerHTML);
-        } catch (error) {
-            return null;
-        }
-    });
+function getParameterCaseInsensitive(object, key) {
+  return object[Object.keys(object).find(k => k.toLowerCase() === key.toLowerCase())];
+}
+
+async function getSpeculationRules() {
+  // Get rules from the HTML
+  const htmlRules = Array.from(document.querySelectorAll('script[type=speculationrules]')).map(script => {
+      try {
+          return JSON.parse(script.innerHTML);
+      } catch (error) {
+          return null;
+      }
+  });
+
+  // Get rules from the HTTP requests
+  const documentRequests = [$WPT_REQUESTS.find( req => req.url === document.location.href)] || [];
```

Comment:
@pmeenan is this the best way of getting the document request? Just `find`ing the first `$WPT_REQUEST` that matches the document.location.href?

One thing that could break is if the page updates the href after fetching (e.g. in SPAs), so ideally there'd be a better way.

Reply:
Went with this instead:

```js
const documentRequest = $WPT_REQUESTS.find(
   req => req.url === performance.getEntriesByType('navigation')[0].name
);
```